Lib/logging/__init__.py

```diff
@@ -339,9 +339,17 @@ def __init__(self, name, level, pathname, lineno,
         self.stack_info = sinfo
         self.lineno = lineno
         self.funcName = func
-        self.created = ct
-        self.msecs = int((ct - int(ct)) * 1000) + 0.0  # see gh-89047
-        self.relativeCreated = (self.created - _startTime) * 1000
+        self.created = ct / 1e9  # ns to float seconds
+
+        # x // 10**n: Drop the rightmost n digits
+        # (x // 10**n) * 10**n: "Change" the rightmost n digits to 0s.
+        # Eg: 55_123_456_789 --> 123_456_789 --> 123_000_000
+        # Lastly divide by 1e6 to change ns to ms
+        # 123_000_000 --> 123
+        # Convert to float by adding 0.0 for historical reasons.
+        self.msecs = (ct - (ct // 1_000_000_000) * 1_000_000_000) // 1_000_000 + 0.0
```

Comment:
It can be simplified:
```suggestion
        self.msecs = (ct % 1_000_000_000) // 1_000_000 + 0.0
```

Reply:
Ah, I never remember to try modulo arithmetic! Thanks, I agree it's much cleaner that way.